WordPress/gutenberg
# Allow to extend the `WP_Theme_JSON_Gutenberg` class

Follow up to https://github.com/WordPress/gutenberg/pull/38625

This PR makes the necessary changes for https://github.com/WordPress/gutenberg/pull/37770 to be able to override the theme.json classes. It extends the requirements of the existing PR to all cases, so we cover future needs as well.  The changes boil down to: make the private functions protected and use `static::` instead of `self::` for accessing variables and functions.

## How to tests

- Verify that unit tests pass.
- Using a block theme load the front and editors. Do some changes in the global styles sidebar within the site editor and save them.
- Using a block theme load the front and editors. Do some changes in the global styles sidebar within the site editor and save them.
